src/DGtal/images/ImageContainerByITKImage.ih

```diff
@@ -200,10 +199,124 @@ namespace DGtal
 
       out << "[ImageContainerByITKImage] valuetype=" << sizeof(TValue) << "bytes "
           << "domain=" << this->domain() << " "
+          << "domainShift=" << this->getDomainShift() << " "
           << "refcount=" << myITKImagePointer->GetReferenceCount() << " "
           << "region=" << region.GetIndex() << "/" << region.GetSize();
     }
 
+    // ----------------- DomainPoint to/from IndexPoint interface -------------
+    template<typename TDomain, typename TValue>
+    inline
+    typename ImageContainerByITKImage<TDomain, TValue>::Point
+    ImageContainerByITKImage<TDomain, TValue>::getDomainPointFromIndex ( const Point & indexPoint ) const
+    {
+      Point dgtal_domain_point = indexPoint;
+      for (Dimension k = 0; k < dimension; k++)
+        dgtal_domain_point[k] = indexPoint[k] + myDomainShift[k];
+
+      return dgtal_domain_point;
```

Comment:
Why not simply:
```C++
{
  return indexPoint + myDomainShift;
}
```

(same for other methods)

Reply:
Nice catch, I can only do this in the functions that do not perform a change of type or mix ITK and DGtal types.

Applied to `getDomainFromIndex` and `getIndexFromDomain`.
Thanks